src/containers/begrunnelse-modal/BegrunnelseModal.tsx

```diff
@@ -0,0 +1,93 @@
+import { Button, Modal, Select } from "@navikt/ds-react";
+import {RefObject, useState} from "react";
+import toast from "react-hot-toast";
+import {useSWRConfig} from "swr";
+import styles from './BegrunnelseModal.module.css';
+
+interface BegrunnelseModalProps {
+    behandlingid: string;
+    modalRef: RefObject<HTMLDialogElement>;
+}
+
+const begrunnelseAlternativer = [
+    "Vedtaksperioden som er foreslått er feil",
+    "Antall dager per uke er feil",
+    "Feil i vilkåret om forholdet til andre ytelser",
+    "Feil i vilkåret om tiltaksdeltagelse",
+    "Se/endre vilkår om barnetillegg",
+    "Returneres etter ønske av saksbehandler",
+]
+
+const BegrunnelseModal = ({behandlingid, modalRef} : BegrunnelseModalProps) => {
+    const mutator = useSWRConfig().mutate;
+    const [begrunnelse, setBegrunnelse] = useState<string>('');
+    const [error, setError] = useState<string>('')
+
+    const lukkModal = () => {
+        modalRef.current?.close();
+    }
+
+    const håndterSendTilbakeMedBegrunnelse = () => {
+        if (begrunnelse == '') {
+            setError('Du må velge en begrunnelse');
+            return;
+        }
+        const res = fetch(`/api/behandling/sendtilbake/${behandlingid}`, {
+            method: 'POST',
+            body: JSON.stringify({begrunnelse: begrunnelse}),
+        }).then(() => {
+           mutator(`/api/behandling/${behandlingid}`).then(() => {
+                toast('Behandling sendt tilbake til saksbehandler');
+            });
+           lukkModal();
+        });
```

Comment:
Hele denne blokken er vel ikke nødvendig når du bruker useSWR? 
Tror du kan skrive under importen: 

const { begrunnelse: string, isError, isLoading, mutate: mutateBegrunnelse } = useSWR<string>('/api/behandling/sendtilbake/${behandlingid')

og så når du skal bruke den kaller du mutateBegrunnelse i selecten

Reply:
Vi må jo hente behandling på nytt for å refreshe siden, og vet ikke om jeg er enig i at det er bedre å erstatte en fetch med en useSWR. Er vel litt mer overhead med sistnevnte.